src/Client.php

```diff
@@ -173,30 +184,23 @@ public function userBalance(string $report_type_uid, bool $detailed = false): Us
     /**
      * {@inheritdoc}
      */
-    public function userReportTypes(bool $can_generate = false,
-                                    bool $content = false,
-                                    string $query = '_all',
-                                    int $size = 20,
-                                    int $offset = 0,
-                                    int $page = 1,
-                                    string $sort = '-created_at',
-                                    bool $calc_total = false): UserReportTypesResponse
+    public function userReportTypes(UserReportTypesParams $params): UserReportTypesResponse
     {
         return UserReportTypesResponse::fromHttpResponse(
             $this->doRequest(new Request('get', 'user/report_types'), [
                 'query' => [
-                    '_can_generate' => $can_generate
+                    '_can_generate' => $params->isCanGenerate()
                         ? 'true'
                         : 'false',
-                    '_content'      => $content
+                    '_content'      => $params->isIncludeContent()
                         ? 'true'
                         : 'false',
-                    '_query'        => $query,
-                    '_size'         => \max(1, $size),
-                    '_offset'       => \max(0, $offset),
-                    '_page'         => \max(1, $page),
-                    '_sort'         => $sort,
-                    '_calc_total'   => $calc_total
+                    '_query'        => $params->getQuery(),
+                    '_size'         => \max(1, $params->getPerPage()),
+                    '_offset'       => \max(0, $params->getOffset()),
+                    '_page'         => \max(1, $params->getPage()),
```

Comment:
@tarampampam @eldario

Potentially, we can transmit an incorrect combination of `_size`, `_offset` & `_page`. Check combination -- Is this `b2bapi` responsibility?

Reply:
This is not the area of b2b-client responsibility